File: modulemain/servergateway/src/main/java/com/redsun/server/gateway/service/UserRolesService.java

package com.redsun.server.gateway.service;

import java.util.List;
import java.util.Map;

import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;

import com.redsun.server.gateway.model.UserRoles;
import com.redsun.server.gateway.model.common.SearchCriteria;

public interface UserRolesService {

	/**
	 * Saves (create or update) the given entity <br>
	 * Transactional operation ( begin transaction and commit )
	 * @param userRoles
	 * @return
	 */
	UserRoles save(UserRoles userRoles);

	/**
	 * Saves (create or update) the given username <br>
	 * Transactional operation ( begin transaction and commit )
	 * @param userRoles
	 * @return
	 */
	List<UserRoles> saveWithUsername(String username, List<Integer> idroles);


	/**
	 * Create the given entity <br>
	 * Transactional operation ( begin transaction and commit )
	 * @param userRoles
	 * @return
	 */
	UserRoles create(UserRoles userRoles);

	/**
	 * Deletes the entity by its Primary Key <br>
	 * Transactional operation ( begin transaction and commit )
	 * @param id
	 */
	void deleteById(Integer id);

	/**
	 * Deletes the entity by username <br>
	 * Transactional operation ( begin transaction and commit )
	 * @param username
	 */
	void deleteByUsername(String username);
	
	/**
	 * Loads the entity for the given Primary Key <br>
	 * @param id
	 * @return the entity loaded (or null if not found)
	 */
	UserRoles getById(Integer id);

	/**
	 * Loads the entity for the given username <br>
	 * @param username
	 * @return the entity loaded (or null if not found)
	 */
	List<UserRoles> listByUsername(String username);

	/**
	 * Loads ALL the entities (use with caution)
	 * @return
	 */
	List<UserRoles> listAll();

	/**
	 * Count all the occurrences
	 * @return
	 */
	long countAll();

	/**
	 * Check exist
	 * @return
	 */
	boolean isExist(Integer id);

	/**
	 * Search by multiple criteria by page.
	 * @return
	 */
	Page<UserRoles> listWithCriterasByPage(List<SearchCriteria> searchCriterias, Pageable pageable);
	
	/**
	 * Loads all roles.
	 * @return
	 */
	List<Map<String, Object>> listAllRoles();

	/**
	 * Loads roles for the given username <br>
	 * @param username
	 * @return the entity loaded (or null if not found)
	 */
	List<Integer> listIdRolesByUsername(String username);
	
}

File: modulemain/clientmain/src/main/resources/static/js/service/userrolesService.js


/**
 * Service for Userroles
 **/

define(['require', 'angular'], function (require, angular) {
app.aService(clientmain.prefix + 'userrolesService', function($http, $rootScope) {
	
	// Save.
	this.saveWithUsername = function(username, idroles) {
		var serverUrl = clientmain.authenUrl + '/userroles/saveWithUsername/' + username;
		var request = {
				 method: 'POST',
				 url: serverUrl,
				 data: {idroles: idroles}
				}
		return $http(request);
	}
	
	// Create.
	this.create = function(userroles) {
		var serverUrl = clientmain.authenUrl + '/userroles/create';
		var request = {
				 method: 'POST',
				 url: serverUrl,
				 data: userroles
				}
		return $http(request);
	}
	
	// Delete.
	this.delete = function(id) {
		var serverUrl = clientmain.authenUrl + '/userroles/delete/' + id;
		var request = {
				 method: 'DELETE',
				 url: serverUrl
				}
		return $http(request);
	}
	
	// Get by Id.
	this.getById = function(userrolesId) {
		var serverUrl = clientmain.authenUrl + '/userroles/getById/' + userrolesId;
		var request = {
				 method: 'GET',
				 url: serverUrl
				}
		return $http(request);
	}
	
	// List all roles.
	this.listAllRoles = function() {
		var serverUrl = clientmain.authenUrl + '/userroles/listAllRoles';
		var request = {
				 method: 'GET',
				 url: serverUrl
				}
		return $http(request);
	}
	
	// List idroles by username.
	this.listIdRolesByUsername = function(username) {
		var serverUrl = clientmain.authenUrl + '/userroles/listIdRolesByUsername/' + username;
		var request = {
				 method: 'GET',
				 url: serverUrl
				}
		return $http(request);
	}

});

});
